### PR TITLE
add dependencies checker to the release workflow

### LINKED ISSUE #182: Add a github action for licensing compliance checker
https://github.com/marketplace/actions/dependencies-license-compliance-checker

### PR DESCRIPTION
**Pull request by stan-dot**
Fixes #182

### CODE REVIEW

**Comment by stan-dot** [comment]
no idea why is this failing, the change in this PR was to the `release` workflow logic, which was not even ran here

```
FAILED tests/test_example.py::test_example_repo_updates - AssertionError: You are downgrading from 2.3.0 to 2.1.0.post31.dev0+56b5bde. Downgrades are not supported.
  
assert 1 == 0
 +  where 1 = CompletedProcess(args=['copier', 'update', '--trust', '--vcs-ref=HEAD', '--data-file', '/home/runner/work/python-copier-template/python-copier-template/example-answers.yml'], returncode=1, stdout=b'You are downgrading from 2.3.0 to 2.1.0.post31.dev0+56b5bde. Downgrades are not supported.\n').returncode
 
 ```

**Comment by coretl** [comment]
> no idea why is this failing, the change in this PR was to the `release` workflow logic, which was not even ran here
> 
> ```
> FAILED tests/test_example.py::test_example_repo_updates - AssertionError: You are downgrading from 2.3.0 to 2.1.0.post31.dev0+56b5bde. Downgrades are not supported.
>   
> assert 1 == 0
>  +  where 1 = CompletedProcess(args=['copier', 'update', '--trust', '--vcs-ref=HEAD', '--data-file', '/home/runner/work/python-copier-template/python-copier-template/example-answers.yml'], returncode=1, stdout=b'You are downgrading from 2.3.0 to 2.1.0.post31.dev0+56b5bde. Downgrades are not supported.\n').returncode
>  
> ```

Part of the CI updates the example repo, so you need to rebase any PRs on main. This is 39 commits behind main at the moment, which means it tries to downgrade the example repo as part of the checks

**Comment by coretl** [comment]
Please also provide:
- A reason why we should add this
- What prompted you to add this
- A project where it is in use and providing useful output
- Details on whether it is checking just the project requirements or also the dev extra requirements

**Comment by stan-dot** [comment]
I just added 'fixes issue' 

https://github.com/DiamondLightSource/python-copier-template/issues/182

File: .github/workflows/_release.yml [hunk 1]
```diff
@@ -11,6 +11,9 @@ jobs:
         with:
           merge-multiple: true
 
+      - name: Dependencies license compliance checker
+        uses: pilosus/action-pip-license-checker@v2.0.0
```
Comment (GDYendell) [inline .github/workflows/_release.yml:15]: I think this should pin to a SHA, as this isn't a trusted source

Comment (stan-dot) [reply .github/workflows/_release.yml:15]: looking into it now

Comment (stan-dot) [reply .github/workflows/_release.yml]: pushed

**Comment by stan-dot** [comment]
ok

**Comment by stan-dot** [comment]
https://github.com/DiamondLightSource/dodal/issues/825

**Pull request closed without merging**